dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/event/TrackedEntityAttributeController.java

```diff
@@ -223,4 +225,29 @@ private Map<String, String> getRequiredValues( TrackedEntityAttribute attr, Map<
         return result;
     }
 
+    // --------------------------------------------------------------------------
+    // Hooks
+    // --------------------------------------------------------------------------
+    @Override
+    protected void forceFiltering( final WebOptions webOptions, final List<String> filters )
+    {
+        if ( webOptions == null || !webOptions.isTrue( "indexableOnly" ) )
+        {
+            return;
+        }
+
+        if ( filters.stream().anyMatch( f -> f.startsWith( "id:" ) ) )
+        {
+            throw new IllegalArgumentException(
+                "indexableOnly parameter cannot be set if a separate filter for id is specified" );
+        }
+
+        StringBuilder sb = new StringBuilder();
+        sb.append( "id:in:" );
+        Set<TrackedEntityAttribute> indexableTeas = trackedEntityAttributeService
+            .getAllTrigramIndexableTrackedEntityAttributes();
```

Comment:
could this return an empty set? and if so would the filter that we are constructing here be valid and its behavior be ok for the user? I haven't seen a test for this case (which does not mean that there is none 😋 ).

Reply:
In theory it shouldnt return an empty set. But to make it cleaner, its better to handle it. Good catch! Also added a test case for that.